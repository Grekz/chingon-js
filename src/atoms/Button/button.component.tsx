import styled from '@emotion/styled'
import * as React from 'react'
interface IButton {
  color?: string
}

type ButtonType = React.SFC<IButton>

const WrapperButton: ButtonType = styled.button`
  color: ${(props: IButton) => props.color};
  background: black;
`

const Button: ButtonType = props => <WrapperButton>{props}</WrapperButton>

export default Button
