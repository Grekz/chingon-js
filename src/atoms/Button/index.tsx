export { default, ButtonType } from './button.component'
