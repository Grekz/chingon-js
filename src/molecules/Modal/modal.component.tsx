import styled from '@emotion/styled'
import * as React from 'react'
import Text from '../../atoms/Text'

interface IModalProps {
  header?: string
  children: React.ReactNode
  footer?: React.ReactNode
  isShown?: boolean
}

const Modal = ({ header, children, footer }: IModalProps) => {
  return (
    <ModalWrapper>
      <ModalOverlay />
      <ModalContent
        aria-modal={true}
        aria-hidden={true}
        tabIndex={-1}
        role="dialog"
      >
        <ModalHeader>
          <Text type="h3">{header}</Text>
          <button type="button" aria-label="Close">
            <span aria-hidden="true">&times;</span>
          </button>
        </ModalHeader>
        <ModalBody>{children}</ModalBody>
        {footer ? <ModalFooter>{footer}</ModalFooter> : null}
      </ModalContent>
    </ModalWrapper>
  )
}

const ModalWrapper = styled.div`
  z-index: 100;
`

const ModalOverlay = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  overflow: hidden;
  outline: 0;
  overflow-x: hidden;
  overflow-y: auto;
  background-color: rgba(0, 0, 0, 0.3);
`

const ModalContent = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  width: 100%;
  pointer-events: auto;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid rgba(0, 0, 0, 0.2);
  border-radius: 0.3rem;
  outline: 0;
`

const ModalHeader = styled.div`
  display: flex;
  align-items: flex-start;
  justify-content: space-between;
  padding: 1rem 1rem;
  border-bottom: 1px solid #dee2e6;
  border-top-left-radius: 0.3rem;
  border-top-right-radius: 0.3rem;
  h3 {
    margin: 0;
  }
  button {
    float: right;
    font-size: 1.5rem;
    font-weight: 700;
    line-height: 1;
    color: #000;
    text-shadow: 0 1px 0 #fff;
    opacity: 0.5;
    padding: 0;
    background-color: transparent;
    border: 0;
    appearance: none;
  }
`

const ModalBody = styled.div`
  position: relative;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  padding: 1rem;
`

const ModalFooter = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
  padding: 1rem;
  border-top: 1px solid #dee2e6;
  border-bottom-right-radius: 0.3rem;
  border-bottom-left-radius: 0.3rem;
`

export default Modal
