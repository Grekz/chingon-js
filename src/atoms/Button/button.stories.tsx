// @flow
import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'
import Button from '.'

storiesOf('Atoms|Button', module).add('As Default', () => (
  <Button>Some button</Button>
))
