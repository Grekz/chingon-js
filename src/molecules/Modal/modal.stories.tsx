import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import Modal from '.'
import Button from '../../atoms/Button'
import { ButtonTypes } from '../../atoms/Button/button.component'

storiesOf('Molecules|Modal', module).add('As Default', () => (
  <Modal
    header="Hello modal"
    footer={
      <>
        <Button buttonType={ButtonTypes.SECONDARY}>Cancel</Button>&nbsp;
        <Button buttonType={ButtonTypes.PRIMARY}>Accept</Button>
      </>
    }
  >
    <p>Modal</p>
  </Modal>
))
