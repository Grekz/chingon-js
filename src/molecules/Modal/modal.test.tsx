import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import serializer, { matchers } from 'jest-emotion'
import * as React from 'react'

import Modal from '.'
import Button from '../../atoms/Button'
import { ButtonTypes } from '../../atoms/Button/button.component'

configure({ adapter: new Adapter() })
expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

describe('<List /> rendering', () => {
  it('should render List', () => {
    const wrapper = shallow(
      <Modal
        header="Hello modal"
        footer={
          <>
            <Button buttonType={ButtonTypes.SECONDARY}>Cancel</Button>&nbsp;
            <Button buttonType={ButtonTypes.PRIMARY}>Accept</Button>
          </>
        }
      >
        <p>Modal</p>
      </Modal>
    )
    expect(wrapper).toHaveLength(1)
  })
  it('should match snapshot', () => {
    const tree = shallow(
      <Modal
        header="Hello modal"
        footer={
          <>
            <Button buttonType={ButtonTypes.SECONDARY}>Cancel</Button>&nbsp;
            <Button buttonType={ButtonTypes.PRIMARY}>Accept</Button>
          </>
        }
      >
        <p>Modal</p>
      </Modal>
    )
    expect(toJson(tree)).toMatchSnapshot()
  })
})
