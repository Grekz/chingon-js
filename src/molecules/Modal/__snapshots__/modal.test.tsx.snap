// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> rendering should match snapshot 1`] = `
<Styled(div)>
  <Styled(div) />
  <Styled(div)
    aria-hidden={true}
    aria-modal={true}
    role="dialog"
    tabIndex={-1}
  >
    <Styled(div)>
      <Text
        color="#000"
        type="h3"
      >
        Hello modal
      </Text>
      <button
        aria-label="Close"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
    </Styled(div)>
    <Styled(div)>
      <p>
        Modal
      </p>
    </Styled(div)>
    <Styled(div)>
      <Button
        buttonType="secondary"
      >
        Cancel
      </Button>
       
      <Button
        buttonType="primary"
      >
        Accept
      </Button>
    </Styled(div)>
  </Styled(div)>
</Styled(div)>
`;
